cmd/object-api-utils.go

```diff
@@ -771,7 +771,7 @@ type snappyCompressReader struct {
 
 func newSnappyCompressReader(r io.Reader) *snappyCompressReader {
 	cr := &snappyCompressReader{r: r}
-	cr.w = snappy.NewBufferedWriter(&cr.buf)
+	cr.w = snappy.NewWriter(&cr.buf)
```

Comment:
I think it should be communicated that a downgrade of the server will not be possible after this has been applied.

I don't know if there is any promise of this, but noting it somewhere will probably be a good idea.

Reply:
> I don't know if there is any promise of this, but noting it somewhere will probably be a good idea.

Downgrades are never allowed in MinIO in general - I will send a separate change to migrate to s2 for existing compression implementation.

Reply:
But we mention that as a breaking change in our Release notes.

Reply:
👍

Reply:
@harshavardhana It is not required to convert unless they want the compression increase applied. It should decode existing snappy content without problems.

Reply:
> @harshavardhana It is not required to convert unless they want the compression increase applied. It should decode existing snappy content without problems.

Yeah agreed there is no migration of existing data but simply new data is written in the newer format @klauspost